src/OSPSuite.Presentation/Binders/ObservedDataDragDropBinder.cs

```diff
@@ -80,6 +100,38 @@ private IReadOnlyList<ObservedDataNode> observedDataNodesFromClassificationNodes
          return observedDataList;
       }
 
+      private Cache<string, List<DataRepository>> observedDataNodesWithFolderPathFromClassificationNodes(IEnumerable<ITreeNode> treeNodes)
+      {
+         var observedDataWithFolderAddressCache = new Cache<string, List<DataRepository>>();
+         treeNodes.Each(treeNode =>
+         {
+            var classificationNode = getClassificationNodeFrom(treeNode);
+            classificationNode.AllLeafNodes.OfType<ObservedDataNode>().Each(observedDataNode =>
+            {
+               if (observedDataWithFolderAddressCache.Contains(observedDataNode.ParentNode.Id))
+                  observedDataWithFolderAddressCache[observedDataNode.ParentNode.Id].Add(observedDataNode.Tag.Subject);
+               else
+                  observedDataWithFolderAddressCache.Add(observedDataNode.ParentNode.Id, new List<DataRepository> { observedDataNode.Tag.Subject });
+            });
+         });
+         return observedDataWithFolderAddressCache;
+      }
+
+      //when dragging and dropping just a selection of observed data and not folders, each one should get a different color
+      private static Cache<string, List<DataRepository>> observedDataNodesWithFolderPathFromObservedDataNodes(IEnumerable<ITreeNode> treeNodes)
+      {
+         var observedDataWithFolderAddressCache = new Cache<string, List<DataRepository>>();
+         treeNodes.Each(node =>
```

Comment:
treeNode.OfType<ObservedDataNode>(). Each....
That way you don't have to cast for type

Reply:
way simpler, thanks :) Corrected